### PR TITLE
Use context of node manager feature as environment variables

### PR DESCRIPTION
**Pull request by mhmxs**

### CODE REVIEW

File: controllers/storageoscluster/node-manager_operand_test.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+			wantPatch:              "[{\"op\":\"add\",\"path\":\"/spec/template/spec/containers/1\",\"value\":{\"env\":[{\"name\":\"foo\",\"value\":\"\"},{\"name\":\"bar\",\"value\":\"\"},{\"name\":\"NODE_NAME\",\"valueFrom\":{\"fieldRef\":{\"apiVersion\":\"v1\",\"fieldPath\":\"spec.nodeName\"}}}],\"image\":\"dummy-image\",\"livenessProbe\":{\"httpGet\":{\"path\":\"/healthz\",\"port\":8081},\"initialDelaySeconds\":15,\"periodSeconds\":20},\"name\":\"dummy-container\",\"readinessProbe\":{\"httpGet\":{\"path\":\"/readyz\",\"port\":8081},\"initialDelaySeconds\":5,\"periodSeconds\":1}}}]",
+		},
+	}
+
+	for _, tc := range cases {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			patch := generateSideCarContainerPatch(tc.nextIndex, tc.containerName, tc.image, tc.context, tc.readinessProbeInterval)
+
+			var asd interface{}
```
Comment (Mojachieee) [inline controllers/storageoscluster/node-manager_operand_test.go:54]: I'm trying to think of a better name then `asd` but it's quite tricky 😆

**Review by Mojachieee — APPROVED** [review]
Looks good, but it would be nice to get some context on how we plan to use this

**Comment by mhmxs** [comment]
@Mojachieee There is not much context. When i designed the config i wanted to give some space to customize the features. There was no requirement previously, but seems this should be the first use case where we would like pass environment variables to a sidecar. This feature expects the context of the feature is a comma separated list of environment variables.

File: controllers/storageoscluster/node-manager_operand.go [hunk 2]
```diff
@@ -314,8 +315,8 @@ func getNodeManagerBuilder(fs filesys.FileSystem, obj client.Object, kcl kubectl
 	nextIndex := 2 // Needs to update if number of containers changes in deployment.
 
 	// Append upgrade guard as sidecar.
-	if _, ok := cluster.Spec.NodeManagerFeatures[upgradeGuardFeatureKey]; ok {
-		mutator := getSideCarContainerMutator(nextIndex, cluster.Spec.Images.UpgradeGuardContainer, os.Getenv(kubeUpgradeGuardEnvVar), "upgrade-guard", upgradeGuardReadinessProbeInterval)
+	if context, ok := cluster.Spec.NodeManagerFeatures[upgradeGuardFeatureKey]; ok {
```
Comment (nolancon) [inline controllers/storageoscluster/node-manager_operand.go:318]: Is `context` the best name for this? just confused me a little bit on initial read through with actual go `context` package

**Review by nolancon — APPROVED** [review]

**Pull request merged**